src/python/pants/backend/python/subsystems/setup.py

```diff
@@ -208,6 +215,21 @@ class PythonSetup(Subsystem):
         ),
         advanced=True,
     )
+    resolves_generate_lockfiles = BoolOption(
+        "--resolves-generate-lockfiles",
+        default=True,
+        help=(
+            "If False, Pants will not attempt to generate lockfiles for `[python].resolves` when "
+            "running the `generate-lockfiles` goal.\n\n"
+            "This is intended to allow you to manually generate lockfiles as a workaround for the "
+            "issues described in the `[python].enable_resolves` option.\n\n"
+            "If you set this to False, Pants will not attempt to validate the metadata headers "
+            "for your user lockfiles. This is useful so that you can keep "
+            "`[python].invalid_lockfile_behavior` to `error` or `warn` if you'd like so that tool "
+            "lockfiles continue to be validated, while user lockfiles are skipped."
```

Comment:
I don't think so. `--invalid-lockfile-behavior` only controls whether we error or not for invalid lockfiles - it should not control whether the lockfile is generated in the first place.

Also I'm envisioning that this will be a temporary flag and get deprecated in 2.11 onc we can switch to Pex for lockfile generation. It would be harder to deprecate the enum option.